streams/src/main/java/org/apache/kafka/streams/state/InMemoryLRUCacheStore.java

```diff
@@ -0,0 +1,295 @@
+     * <strong>NOTE:</strong> the default serializers and deserializers in the context <em>must</em> match the key and value types
+     * used as parameters for this key value store.
+     * 
+     * @param name the name of the store, used in the name of the topic to which entries are recorded
+     * @param capacity the maximum capacity of the LRU cache; should be one less than a power of 2
+     * @param context the processing context
+     * @param time the time provider; may not be null
+     * @return the key-value store
+     */
+    public static <K, V> InMemoryLRUCacheStore<K, V> create(String name, int capacity, ProcessorContext context, Time time) {
+        return create(name, capacity, context, new Serdes<K, V>(name, context), time);
+    }
+
+    /**
+     * Create an in-memory key value store that records changes to a Kafka topic, the provided serializers and deserializers, and
+     * the system time provider.
+     * 
+     * @param name the name of the store, used in the name of the topic to which entries are recorded
+     * @param capacity the maximum capacity of the LRU cache; should be one less than a power of 2
+     * @param context the processing context
+     * @param keySerializer the serializer for keys; may not be null
+     * @param keyDeserializer the deserializer for keys; may not be null
+     * @param valueSerializer the serializer for values; may not be null
+     * @param valueDeserializer the deserializer for values; may not be null
+     * @return the key-value store
+     */
+    public static <K, V> InMemoryLRUCacheStore<K, V> create(String name, int capacity, ProcessorContext context,
+                                                            Serializer<K> keySerializer, Deserializer<K> keyDeserializer,
+                                                            Serializer<V> valueSerializer, Deserializer<V> valueDeserializer) {
+        return create(name, capacity, context, keySerializer, keyDeserializer, valueSerializer, valueDeserializer, new SystemTime());
+    }
+
+    /**
+     * Create an in-memory key value store that records changes to a Kafka topic, the provided serializers and deserializers, and
+     * the given time provider.
+     * 
+     * @param name the name of the store, used in the name of the topic to which entries are recorded
+     * @param capacity the maximum capacity of the LRU cache; should be one less than a power of 2
+     * @param context the processing context
+     * @param keySerializer the serializer for keys; may not be null
+     * @param keyDeserializer the deserializer for keys; may not be null
+     * @param valueSerializer the serializer for values; may not be null
+     * @param valueDeserializer the deserializer for values; may not be null
+     * @param time the time provider; may not be null
+     * @return the key-value store
+     */
+    public static <K, V> InMemoryLRUCacheStore<K, V> create(String name, int capacity, ProcessorContext context,
```

Comment:
@rhauch I am wondering if this class could not benefit from a static ``Builder`` class. Something along the lines of 
```
InMemoryLRUCacheStore cacheStore = InMemoryLRUCacheStore.Builder.create()
                                                                .name(name)
                                                                .capacity(capacity)
                                                                .context(context)
                                                                .keySerDe(keySerializer, keyDeserializer)
                                                                .valueSerDe(valueSerializer, valueDeserializer)
                                                                .time(time)
                                                                .build();
```

OTOH, the constructor arguments are not the same type (say, String), so no good chances of inverting parameter order or so. Wdyt?